CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFile.swift

```diff
@@ -57,25 +64,38 @@ final class CEWorkspaceFile: Codable, Comparable, Hashable, Identifiable, Editor
     /// Returns the `id` in ``EditorTabID`` enum form
     var tabID: EditorTabID { .codeEditor(id) }
 
-    /// Returns a boolean that is true if ``children`` is not `nil`
-    var isFolder: Bool { url.hasDirectoryPath }
+    /// Returns a boolean that is true if the resource represented by this object is a directory.
+    lazy var isFolder: Bool = {
+        (try? url.resourceValues(forKeys: [.isDirectoryKey]))?.isDirectory == true
+    }()
```

Comment:
Perhaps having a `isFile` would be handy for functions like: `childrenDescription`?

Reply:
It could be handy, but since `childrenDescription` is only for debugging purposes it didn't feel necessary.

Reply:
Great!